lib/translation/post_detection_text.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module DiscourseAi
+  module Translation
+    class PostDetectionText
+      def self.get_text(post)
+        return if post.blank?
+        cooked = post.cooked
+        return if cooked.blank?
+
+        doc = Nokogiri::HTML5.fragment(cooked)
+        original = doc.text.strip
+
+        # quotes and blockquotes
+        doc.css("blockquote, aside.quote").remove
+        # image captions
+        doc.css(".lightbox-wrapper").remove
+
+        necessary = doc.text.strip
+
+        # oneboxes (external content)
+        doc.css("aside.onebox").remove
+        # code blocks
+        doc.css("code, pre").remove
+        # hashtags
+        doc.css("a.hashtag-cooked").remove
+        # emoji
+        doc.css("img.emoji").remove
+        # mentions
+        doc.css("a.mention").remove
```

Comment:
I would group stuff in constants, to have only two remove calls.

I don't think the comments are adding much: `# emoji` => `img.emoji`. But if you still think they are important you can format like this:

```
NECESSARY_SELECTORS = [
  "aside.onebox", # oneboxes (external content)
  "code, pre", # code blocks
  ...
]
```

and then call: `doc.css(*NECESSARY_SELECTORS).remove`

Reply:
I would also add that necessary/preferred is not very clear. I don't really know why they are necessary or preferred, this is where we would need a comment IMO.